server/src/main/java/org/apache/lucene/util/LongHashSet.java

```diff
@@ -30,8 +30,12 @@ public final class LongHashSet implements Accountable {
     /** maximum value in the set, or Long.MIN_VALUE for an empty set */
     public final long maxValue;
 
-    /** Construct a set. Values must be in sorted order. */
     public LongHashSet(long[] values) {
```

Comment:
@Shailesh-Kumar-Singh Thanks for the fix, I think the cleanest (but a bit more verbose approach) would be to clone `LongHashSet` as `UnsignedLongHashSet`:
 - `MISSING` has to be set to `Numbers.MIN_UNSIGNED_LONG_VALUE_AS_LONG`
 - `minValue` value bound has to be  `Numbers.MIN_UNSIGNED_LONG_VALUE_AS_LONG`
- `maxValue` value bound has to be  `Numbers.MAX_UNSIGNED_LONG_VALUE_AS_LONG`

I actually don't know why we copied the Lucene class, @harshavamsi could you share the context please?

Reply:
We had to bring in the class because it was package private in lucene @reta

Reply:
Thanks @harshavamsi , I think we could drop it now, in 10.1 it is public (+ public constructors):

```
public class LongHashSet implements Iterable<LongCursor>, Accountable, Cloneable {
```

Reply:
Oh, there are few of them, `org.apache.lucene.document.LongHashSet`, it is gone in 10.x